script.js

```diff
@@ -1,11 +1,16 @@
+const TODOS_SESSION_KEY = "todos";
+
+const storedTodos = JSON.parse(sessionStorage.getItem(TODOS_SESSION_KEY));
+const initialState = {
+  todos: storedTodos || [],
+  filter: "all",
+};
+
 const form = document.getElementById("todoForm");
 const input = document.getElementById("inputField");
 const list = document.getElementById("list");
 
-const state = {
-  todos: [],
-  filter: "all",
-};
+const state = { ...initialState };
 
```

Comment:
You need to render the todos list here simply by invoking `renderTodos()`.  After refreshing the list vanishes from the page and appears only after switching the filter. This is because event listener for `change` event invokes `renderTodos()` which does the job.